modules/24_trade/se_trade/bounds.gms

```diff
@@ -0,0 +1,129 @@
+        );
+$ENDIF.fossil_realization
+    );
+);
+
+
+*** bounds on oil, gas and coal trade in 2010 and 2015 (+-10% from historical IEA values)
+loop( ttot$(ttot.val eq 2010 OR ttot.val eq 2015),
+	loop( peFos,
+		vm_Xport.lo(ttot,regi,peFos) = 0.9 * pm_IO_trade(ttot,regi,peFos,"Xport");
+		vm_Xport.up(ttot,regi,peFos) = 1.1 * pm_IO_trade(ttot,regi,peFos,"Xport");
+		vm_Mport.lo(ttot,regi,peFos) = 0.9 * pm_IO_trade(ttot,regi,peFos,"Mport");
+		vm_Mport.up(ttot,regi,peFos) = 1.1 * pm_IO_trade(ttot,regi,peFos,"Mport");
+	);
+);
+
+*** trade upper bound causes a infeasibility in REMIND-EU. Using national pe (even after Mports adjustment in trade module) is impossible due to the lack of reserves (vm_fuExtr maximum bound) causing either a infeasibsibility or pushing the national extraction to a point in the supply curve of very high prices. The below code relax the historical trade bounds in this region, assuming much more flexible trade, to levelise the pe fossil availability and supply prices. The allowed increase imports fixing the feasibility issue and the allowed increase exports allow more trade within EU if necessary to help in the price issue. Alternatively, the model should approximate better national fossil reserves (compatibility of other sources with IEA data), adjust capacity factor variation between years (dot and refliq for oil; ... for coal), and/or better approximate historical capacities in 2005, 2010 and 2015.
+loop( regi$regi_group("EUR_regi",regi),
+	loop(ttot$(ttot.val eq 2010 OR ttot.val eq 2015),
+		loop( peFos,
+			vm_Xport.lo(ttot,regi,peFos) = 0.5 * pm_IO_trade(ttot,regi,peFos,"Xport");
+			vm_Xport.up(ttot,regi,peFos) = 1.5 * pm_IO_trade(ttot,regi,peFos,"Xport");
+			vm_Mport.lo(ttot,regi,peFos) = 0.5 * pm_IO_trade(ttot,regi,peFos,"Mport");
+			vm_Mport.up(ttot,regi,peFos) = 1.5 * pm_IO_trade(ttot,regi,peFos,"Mport");
+		);
+	);
+);
+
+*** upper bounds ( 1% yearly growth rate) on all big oil exporters (more than 15EJ in 2010) in 2020, 2025 and 2030
+loop(regi,
+      if( (pm_IO_trade("2010",regi,"peoil","Xport") ge (15*sm_EJ_2_TWa)),
+        vm_Xport.up("2020",regi,"peoil") = ((1 + 0.02) **  10) * pm_IO_trade("2010",regi,"peoil","Xport");
+        vm_Xport.up("2025",regi,"peoil") = ((1 + 0.02) **  15) * pm_IO_trade("2010",regi,"peoil","Xport");
+        vm_Xport.up("2030",regi,"peoil") = ((1 + 0.02) **  20) * pm_IO_trade("2010",regi,"peoil","Xport");
+      );
+);
+
+*** FS: constrain biomass imports in EU subregions from cm_startyear or 2020 onwards to a quarter of 2015 PE bioenergy demand
+if ( cm_biotrade_phaseout eq 1,
+	vm_Mport.up(t,regi,"pebiolc")$(t.val ge cm_startyear AND t.val gt 2015 AND regi_group("EUR_regi",regi)) = 
+      pm_demPeBio("2015",regi)$(regi_group("EUR_regi",regi))/4;
+);
+
+*** set maximum import and export secondary energy trade based on trading capacities 
+vm_Mport.up(t,regi,entySe) = sum(regi2,p24_seTradeCapacity(t,regi2,regi,entySe));
```

Comment:
What is the purpose of introducing a trade capacity?

Reply:
Trade capacity limits the total trade potential between two regions. You can think of it as the pipelines transportation limits, regasification infrastructure currently installed and transmission power lines capacities that connect two countries/regions.